pilot/cmd/pilot-agent/status/ready/probe.go

```diff
@@ -72,7 +72,9 @@ func (p *Probe) checkConfigStatus() error {
 			p.lastKnownState.versionStats.LDSVersion)
 	}
 
-	if p.lastKnownState.versionStats.CDSVersion > 0 && p.lastKnownState.versionStats.LDSVersion > 0 {
+	// Envoy seems to not updatig cds version (need to confirm) when partial rejection happens.
+	// Till that is fixed, we should treat LDS success as readiness success as LDS happens last.
+	if p.lastKnownState.versionStats.LDSVersion > 0 {
```

Comment:
@howardjohn for partial cluster rejections, Envoy seems to be not updating this (need to verify though), that is why we are seeing cds as Not Received and lds as Received. For now, I changed it like this - I will verify in Envoy and follow-up